functions/APIs/deals.js

```diff
@@ -0,0 +1,115 @@
+const { db } = require('../util/admin');
+
+exports.getAllDeals = (request, response) => {
```

Comment:
It is a little bit unusual to use arrow functions at top-level like this.  It's not wrong, but I don't usually see it written like this.  Are you following some example or template?

Reply:
I usually use arrow functions for RESTful APIs.  Should I use regular functions instead of arrow functions?

Reply:
I think for top-level functions, using regular functions would be better, yeah.  Arrow functions are technically anonymous functions, so it's sometimes a little harder to debug code that uses them.

Reply:
I found out that the Firebase [documentation](https://firebase.google.com/docs/functions/get-started) uses arrow functions.  Should I use regular functions if the sample code uses arrow functions?

Reply:
Hm, where are you seeing arrow functions in there?  All of the examples there seem to export calls to `functions.https.onRequest`, which takes an anonymous function as an argument.  That's different from exporting an anonymous function.  Has the documentation changed lately?  Should we be using `functions.https.onRequest`?

Reply:
I found out that the Firebase [documentation](https://firebase.google.com/docs/functions/http-events) uses existing Express apps to call functions via HTTP requests.  I incorporated your feedback into my code by using the Express Router to mount routes and passing a full Express app to an HTTP function.